File: app/src/main/java/com/returnnotfound/stain/widget/image/ImageLoaderFactory.java
package com.returnnotfound.stain.widget.image;

public class ImageLoaderFactory {
  private static final Type DEFAULT = Type.GLIDE;

  private static ImageLoader sGlideInstance = new ImageLoaderGlide();

  public static ImageLoader getInstance() {
    return getInstance(DEFAULT);
  }

  public static ImageLoader getInstance(Type type) {
    switch (type) {
      case GLIDE:
      default:
        return sGlideInstance;
    }
  }

  public enum Type {
    GLIDE, PICASSO
  }
}

File: app/src/main/java/com/returnnotfound/stain/widget/image/ImageLoaderFactory.kt
package com.returnnotfound.stain.widget.image

object ImageLoaderFactory {
  private val DEFAULT = Type.GLIDE
  private val sGlideInstance: ImageLoader = ImageLoaderGlide()
  val instance: ImageLoader get() = getInstance(DEFAULT)

  private fun getInstance(type: Type?): ImageLoader {
    return when (type) {
      Type.GLIDE -> sGlideInstance
      else -> sGlideInstance
    }
  }

  enum class Type {
    GLIDE, PICASSO
  }
}